osmnx/osm_xml.py

```diff
@@ -211,10 +219,6 @@ def save_graph_xml(
         table["changeset"] = "1"
         table["timestamp"] = "2017-01-01T00:00:00Z"
 
-    # convert all datatypes to str
-    gdf_nodes = gdf_nodes.applymap(str)
-    gdf_edges = gdf_edges.applymap(str)
-
```

Comment:
Were these lines only necessary for converting `osm_version` to a string? Is there no other edge case where we must convert values to string? I see you added a few other string conversions below: is this more efficient than `applymap(str)`?

Reply:
I think they were necessary because `xml.etree` expects strings only. The API allows you to input int and float types, but when you actually try to write the XML file, it fails saying that types must all be strings. Thus, everything needs to be a string before passing to `xml.etree`, but we need the original types to check for null / NaN.

From what I can see, these changes do not break any other edge case based on the testing I've done locally.